integration/spark2/src/main/scala/org/apache/spark/sql/execution/CastExpressionOptimization.scala

```diff
@@ -129,6 +143,13 @@ object CastExpressionOptimization {
             } else {
               Some(CastExpr(c))
             }
+          case s: ShortType if t.sameType(IntegerType) =>
```

Comment:
almost code is same in each case please try to refactor

Reply:
Refactored